state/store.go

```diff
@@ -385,12 +401,43 @@ func (store dbStore) LoadABCIResponses(height int64) (*tmstate.ABCIResponses, er
 	return abciResponses, nil
 }
 
+// LoadLastABCIResponses loads the ABCIResponses from the most recent height.
+// The height parameter is used to ensure that the response corresponds to the latest height.
+// If not, an error is returned.
+//
+// This method is used for recovering in the case that we called the Commit ABCI
+// method on the application but crashed before persisting the results.
+func (store dbStore) LoadLastABCIResponse(height int64) (*tmstate.ABCIResponses, error) {
+	bz, err := store.db.Get(lastABCIResponseKey)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(bz) == 0 {
+		return nil, errors.New("no last ABCI response has been persisted")
+	}
+
+	abciResponse := new(tmstate.ABCIResponsesInfo)
+	err = abciResponse.Unmarshal(bz)
+	if err != nil {
+		tmos.Exit(fmt.Sprintf(`LoadLastABCIResponses: Data has been corrupted or its spec has
+			changed: %v\n`, err))
```

Comment:
why is this an exit and not a panic?

Reply:
tbh I cannot remember exactly why this was an exit rather than a panic. What would be the reasoning to change it to a panic just so im clear

Reply:
I see that there are 5 such calls throughout the `state/store.go` file and `Exit` has been the convention in this file for more than 5 years now. I'd be very interested in knowing what the reasoning here was originally.

Looking at all the places this is called from, it doesn't seem as though there are any `defer`s prior to where these functions are called, but at first glance it seems a little dangerous to have `Exit`s embedded within the code instead of panics.

It may be worth a follow-up investigation here after this PR - perhaps a codebase-wide audit of where `Exit`'s used instead of panic to see whether it's still applicable, and where it is applicable we should document why it's applicable in comments colocated with the `Exit` calls.